content/200-concepts/100-components/01-prisma-schema/06-relations/420-relation-mode.mdx

```diff
@@ -0,0 +1,176 @@
+}
+```
+
+Some databases, such as MongoDB or [PlanetScale](/guides/database/using-prisma-with-planetscale#differences-to-consider)),
+do not support foreign keys. For this situation, Prisma offers an alternative _relation mode_ where most properties of relations in relational databases are emulated.
+
+When using Prisma Client the behavior of queries is identical or similar, but all referential actions and most referential integrity is actually handled by the Prisma engine under the hood.
+
+<Admonition type="warning">
+  There are significant performance implications to emulating referential
+  integrity in the Prisma client. In cases where the underlying database can
+  handle referential integrity with foreign keys, it is usually the preferred
+  choice.
+</Admonition>
+
+</TopBlock>
+
+## How to set the relation mode in your Prisma schema
+
+To set the relation mode, you need to enable the `referentialIntegrity` preview feature. Add the `referentialIntegrity` feature flag to the `previewFeatures` list in the [`generator`](/reference/api-reference/prisma-schema-reference#generator) block of your Prisma schema file, and add the `relationMode` field to the [`datasource`](/reference/api-reference/prisma-schema-reference#datasource) block:
+
+```prisma file=schema.prisma highlight=4,9;add
+datasource db {
+  provider     = "mysql"
+  url          = env("DATABASE_URL")
+  relationMode = "prisma"
+}
+
+generator js {
+  provider        = "prisma-client-js"
+  previewFeatures = ["referentialIntegrity"]
+}
+```
+
+To change the relation mode, update the `relationMode` field in the `datasource` block.
+
+<Admonition type="warning">
+
+The <inlinecode>relationMode</inlinecode> field was renamed in Prisma version 4.5.0,
+and was previously named <inlinecode>referentialIntegrity</inlinecode>.
+
+</Admonition>
+
+The available options are:
+
+- `foreignKeys`: this handles relations in the database using foreign keys. This is the default option when using all relational database connectors.
+- `prisma`: this emulates relations in the Prisma client. This is the default (and only available) option when using the MongoDB connector. You should also [enable this option](/guides/database/using-prisma-with-planetscale#how-to-enable-emulation-of-referential-integrity) when using the MySQL connector with a PlanetScale database.
+
+<Admonition type="warning">
+
+When using Prisma Migrate, setting the value to <inlinecode>prisma</inlinecode> will remove all previously created foreign keys in the next migration. Setting the value to <inlinecode>foreignKeys</inlinecode> (either directly or by removing the <inlinecode>relationMode</inlinecode> field and using the default option for relational databases) will create foreign keys for all relations in the next migration.
+
+</Admonition>
+
+<!-- TODO: mention preview feature + version info in intro -->
+
+## Handle relations in your relational database with the <inlineCode>foreignKeys</inlineCode> relation mode
```

Comment:
This is very long in the visual page.
But the information is all correct and seems needed.
Unsure what to do here.